lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -133,6 +133,43 @@ void sendObjectPostDelete(const Ufe::SceneItem::Ptr& sceneItem)
 #endif
 }
 
+void sendObjectDestroyed(const Ufe::Path& ufePath)
+{
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    Ufe::Scene::instance().notify(Ufe::ObjectDestroyed(ufePath));
+#else
+    // Unfortunately in Ufe v1 there was no object destroyed notif
+    // and the only delete notifs we have both take a scene item
+    // which we cannot create for the input Ufe path since that
+    // path is no longer valid (it has already been destroyed).
+    // So the only choice we have is to subtree invalidate the
+    // parent which will remove the destroyed item (keeping all
+    // the valid children).
+    auto sceneItem = Ufe::Hierarchy::createItem(ufePath);
+    while (!sceneItem)
+    {
+        auto parentPath = ufePath.pop();
+        sceneItem = Ufe::Hierarchy::createItem(ufePath);
+    }
+    if (sceneItem) {
+        sendObjectPostDelete(sceneItem);
+        sendObjectAdd(sceneItem);
+    }
+#endif
+}
+
+void sendSubtreeInvalidate(const Ufe::SceneItem::Ptr& sceneItem)
+{
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    Ufe::Scene::instance().notify(Ufe::SubtreeInvalidate(sceneItem));
+#else
+    // In Ufe v1 there was no subtree invalidate notif. So we mimic it by sending
+    // delete/add notifs.
+    sendObjectPostDelete(sceneItem);
+    sendObjectAdd(sceneItem);
+#endif
+}
```

Comment:
Two new helpers (like the Add/PostDelete above) that handle Ufe v1 case. Make the code below much simpler (no more ifdef).